certbot-dns-cloudflare/certbot_dns_cloudflare/_internal/dns_cloudflare.py

```diff
@@ -170,22 +177,23 @@ def _find_zone_id(self, domain):
             try:
                 zones = self.cf.zones.get(params=params)  # zones | pylint: disable=no-member
             except CloudFlare.exceptions.CloudFlareAPIError as e:
-                code = int(e)
-                hint = None
-
-                if code == 6003:
-                    hint = ('Did you copy your entire API token/key? To use Cloudflare tokens, '
-                            'you\'ll need python-cloudflare>=2.3.1. You have python-cloudflare {}'
-                            .format(CloudFlare.__version__))
-                elif code == 9103:
-                    hint = 'Did you enter the correct email address?'
-                elif code == 9109:
-                    hint = ('Does your API token have "Zone:Zone:Read" and "Zone:DNS:Edit" '
-                            'permissions for all zones?')
-
-                raise errors.PluginError('Error determining zone_id: {0} {1}. Please confirm that '
-                                         'you have supplied valid Cloudflare API credentials.{2}'
-                                         .format(code, e, ' ({0})'.format(hint) if hint else ''))
+                if 'com.cloudflare.api.account.zone.list' not in str(e):
```

Comment:
I have the same concern as you about whether or not we can rely on this string never changing. If it does, this plugin will break when requesting a cert for a subdomain.

Because of this, I suggest we:

1. Only raise an error here if it's an error code we currently recognize.
2. Log unraised errors through `logger.debug`.
3. If values like `code` are set, include information about the last raised exception in the error message raised at the bottom of this function.